src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -1384,5 +1393,505 @@ void M(bool b)
+{
+  // Code size       38 (0x26)
+  .maxstack  2
+  IL_0000:  newobj     ""C..ctor()""
+  IL_0005:  dup
+  IL_0006:  brtrue.s   IL_000b
+  IL_0008:  pop
+  IL_0009:  br.s       IL_0015
+  IL_000b:  call       ""delegate*<void> C.GetPtr()""
+  IL_0010:  calli      ""delegate*<void>""
+  IL_0015:  ldnull
+  IL_0016:  dup
+  IL_0017:  brtrue.s   IL_001b
+  IL_0019:  pop
+  IL_001a:  ret
+  IL_001b:  call       ""delegate*<void> C.GetPtr()""
+  IL_0020:  calli      ""delegate*<void>""
+  IL_0025:  ret
+}
+");
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+
+            var invocations = tree.GetRoot().DescendantNodes().OfType<ConditionalAccessExpressionSyntax>().ToList();
+            Assert.Equal(2, invocations.Count);
+            foreach (var invocation in invocations)
+            {
+                var type = model.GetTypeInfo(invocation).Type;
+                Assert.Equal(SpecialType.System_Void, type!.SpecialType);
+            }
+        }
+
+        [Fact]
+        public void FunctionPointerTypePatternIsNull()
+        {
+            var source = @"
+using System;
+unsafe class C
+{
+    static void Main()
+    {
+        delegate*<void> ptr = null;
+        Console.WriteLine(ptr is null);
+        Console.WriteLine(ptr is var v);
+    }
+}";
+
+            var comp = CreateCompilationWithFunctionPointers(source, TestOptions.UnsafeReleaseExe);
+            var verifier = CompileAndVerify(comp, verify: Verification.Skipped, expectedOutput: @"
+True
+True");
+            verifier.VerifyIL("C.Main", expectedIL: @"
+{
+  // Code size       19 (0x13)
+  .maxstack  2
+  .locals init (delegate*<void> V_0) //ptr
+  IL_0000:  ldc.i4.0
+  IL_0001:  conv.u
+  IL_0002:  stloc.0
+  IL_0003:  ldloc.0
+  IL_0004:  ldnull
+  IL_0005:  ceq
+  IL_0007:  call       ""void System.Console.WriteLine(bool)""
+  IL_000c:  ldc.i4.1
+  IL_000d:  call       ""void System.Console.WriteLine(bool)""
+  IL_0012:  ret
+}
+");
+
+            comp = CreateCompilationWithFunctionPointers(source, parseOptions: TestOptions.Regular7_3);
+            comp.VerifyDiagnostics(
+                // (7,9): error CS8652: The feature 'function pointers' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         delegate*<void> ptr = null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "delegate*<void>").WithArguments("function pointers").WithLocation(7, 9)
+            );
+        }
+
+        [Fact]
+        public void FunctionPointerTypePatternIsVarParenthesized()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+using System;
+unsafe class C
+{
+    static void Main()
+    {
+        delegate*<void> ptr = null;
+        _ = ptr is var (x);
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (8,20): error CS8521: Pattern-matching is not permitted for pointer types.
```

Comment:
>// (8,20): error CS8521: Pattern-matching is not permitted for pointer types. [](start = 16, length = 77)

Doesn't this error sound misleading, given that ```ptr is var v``` works?

Reply:
Possibly, but changing that is beyond the scope of this PR given that this is the behavior for regular pointers as well. I can file a follow up issue if you'd prefer.

---
In reply to: [408276884](https://github.com/dotnet/roslyn/pull/43248#discussion_r408276884) [](ancestors = 408276884)

Reply:
>  I can file a follow up issue if you'd prefer.

That would be great. 

---
In reply to: [408284161](https://github.com/dotnet/roslyn/pull/43248#discussion_r408284161) [](ancestors = 408284161,408276884)

Reply:
[https://github.com/dotnet/roslyn/issues/43339](https://github.com/dotnet/roslyn/issues/43339)

---
In reply to: [408288492](https://github.com/dotnet/roslyn/pull/43248#discussion_r408288492) [](ancestors = 408288492,408284161,408276884)